pylutron/__init__.py

```diff
@@ -103,8 +103,10 @@ def _do_login_locked(self):
     try:
       sock = self._telnet.get_socket()
       sock.setsockopt(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1)
-      # Send keepalive probes after 60 seconds of inactivity
-      sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_KEEPIDLE, 60)
+      # Detect if system supports KEEP* - Mac and Windows don't
+      if getattr(socket, 'TCP_KEEPIDLE', None) is not None:
+        # Send keepalive probes after 60 seconds of inactivity
+        sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_KEEPIDLE, 60)
```

Comment:
But we do want this functionality on the mac though. What is the right option for the mac?

Reply:
Ah, I think we may want something like:
```python
keepalive_opt = None
# Linux
if getattr(socket, 'TCP_KEEPIDLE', None) is not None:
  keepalive_opt = socket.TCP_KEEPIDLE
# Mac/Windows
elif getattr(socket, 'TCP_KEEPALIVE', None) is not None:
  keepalive_opt = socket.TCP_KEEPALIVE
if keepalive_opt is not None:
  sock.setsockopt(...)
```